### PR TITLE
race condition fix

### PR DESCRIPTION
**Pull request by bheni**
fixes race condition that occurs when flushing on one routine and creating row iterators on another

### CODE REVIEW

File: go/libraries/doltcore/table/table_reader.go [hunk 1]
```diff
@@ -97,6 +97,14 @@ func NewBufferedTableReader(ctx context.Context, tbl *doltdb.Table) (SqlTableRea
 	return newPkTableReader(ctx, tbl, sch, true)
 }
 
+// NewBufferedTableReader creates a buffered SqlTableReader from |tbl| starting from the first record.
+func NewBufferedReaderForRows(ctx context.Context, rowData types.Map, sch schema.Schema) (SqlTableReader, error) {
```
Comment (zachmu) [inline go/libraries/doltcore/table/table_reader.go:101]: Seems to be unused?

**Review by zachmu — APPROVED** [review]
This looks good, but Daylon should take a closer look

**Review by Hydrocharged — APPROVED** [review]
LGTM as well

File: go/libraries/doltcore/table/table_reader.go (hunk 1 above)
Comment (Hydrocharged) [reply go/libraries/doltcore/table/table_reader.go:101]: Also the comment is wrong

File: go/libraries/doltcore/sqle/dtables/branches_table.go [hunk 2]
```diff
@@ -67,7 +68,7 @@ func (bt *BranchesTable) Schema() sql.Schema {
 
 // Partitions is a sql.Table interface function that returns a partition of the data.  Currently the data is unpartitioned.
 func (bt *BranchesTable) Partitions(*sql.Context) (sql.PartitionIter, error) {
-	return sqlutil.NewSinglePartitionIter(), nil
+	return sqlutil.NewSinglePartitionIter(types.Map{}), nil
```
[the remark this answers is not in the document]
Comment (bheni) [reply go/libraries/doltcore/sqle/dtables/branches_table.go:71]: I typically only use EmptyMap if I want the objects to be usable as a map.  In this case using the map data stored within the object won't give the correct result anyway, so constructing an empty map will only hide the fact that the wrong data is being used.

[the remark this answers is not in the document]
Comment (Hydrocharged) [reply go/libraries/doltcore/sqle/dtables/branches_table.go:71]: Makes sense!

**Pull request merged**